src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -57,6 +64,22 @@ public partial class HubConnection
         public TimeSpan ServerTimeout { get; set; } = DefaultServerTimeout;
         public TimeSpan HandshakeTimeout { get; set; } = DefaultHandshakeTimeout;
 
+        // TODO: Documentation
+        public HubConnectionStatus Status
+        {
+            get
+            {
+                // Local copy for thread-safety
+                var connectionState = _connectionState;
```

Comment:
Yes but the field can be set to null by another thread in-between checking it is not null and getting IsStopped.

Copy is the wrong word. A better comment would be "Local reference for thread-safety".

Reply:
Ah. Thanks for the clarification 🙂

Reply:
`?.` will do this for you ([source](https://sharplab.io/#v2:D4AQDABCCMDcCwAoEBmKAmCBhCBvJEhEADgE4CWAbgIYAuAptgPYB2L9AxreawMq11GAfQDOAhgkREIBIqmZtO3PuICuIiPzrq8s6YQDm9Wrqn7z5AGYAKUePoB+AHT8mxYvQAmEBw4i1SVXoASlNzcLkAdgV2Lh4WLVp1F1o3D09JCOkAXz0skGisVljlBLURFOpSBgy8nLzcs0I8+RBMIsU4lUEw8PkAIyYmABtNVPcvPCNaWEb6ppkF+XoWVQBbGKV4xJ18Bektaq8AGjqiVwnPBqRsoA))

It's mostly just a clever trick though, so I don't really mind either way. Was mostly just curious if if would do the Right Thing.

Reply:
I like taking a local variable for thread-safety issues because it gives a good opportunity to comment and make it obvious that it is a thing future readers of the code need to think about.